src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java

```diff
@@ -289,8 +317,7 @@ public void remove() {
 		if (player.isOnline() && !player.isDead()) {
 			player.eject();
 		}
-		player.setAllowFlight(canFly);
-		player.setFlying(hadFly);
+		ProjectKorra.flightHandler.removeInstance(player, "FlightMultiAbility");
```

Comment:
This line will manage the player's flight state, but again change "FlightMultiAbility" to getName() or a final String with the same value.

Reply:
Refer to second comment